contract/src/models/json_types.rs

```diff
@@ -58,6 +80,39 @@ pub struct JsonNFTData {
     pub contract_id: AccountId,
 }
 
+/// Public Sale Data that is serializable
+#[derive(BorshDeserialize, BorshSerialize, Serialize, Deserialize, Clone)]
+#[serde(crate = "near_sdk::serde")]
+pub struct JsonPublicSaleConfig {
+    /// Maximum number of keys that can be added to this drop. If None, there is no max.
+    pub max_num_keys: Option<u64>,
+ 
+    /// Amount of $NEAR that the user needs to attach (if they are not the funder) on top of costs. This amount will be
+    /// Automatically sent to the funder's balance. If None, the keys are free to the public.
+    pub price_per_key: Option<U128>,
+ 
+    #[serde(skip_serializing)]
+    /// Which accounts are allowed to add keys?
+    pub allowlist: Option<Vec<AccountId>>,
+ 
+    #[serde(skip_serializing)]
+    /// Which accounts are NOT allowed to add keys?
```

Comment:
Should skip serialization be the default behavior if options are null? It's either passes as null or left undefined on the returned object so... Same thing for client devs?